src/components/App.tsx

```diff
@@ -1,6 +1,121 @@
+  unit: Units;
+  newCityInput: string;
+
+  weatherInfo: TWeather | null;
+  locationErr: boolean;
+  weatherErr: boolean;
+  isLocationLoading: boolean;
+  isWeatherLoading: boolean;
+  error: string;
+}
+
+export class App extends Component<{}, AppState> {
+  state: AppState = {
+    city: '',
+    unit: 'metric',
+    newCityInput: '',
+    weatherInfo: null,
+    locationErr: false,
+    weatherErr: false,
+    isLocationLoading: true,
+    isWeatherLoading: true,
+    error: '',
+  };
+
+  localStorageKey = 'city';
+
+  debouncedSearch = debounce((newCityInput: AppState['newCityInput']) => {
+    this.getWeatherData(newCityInput || this.state.city, this.state.unit);
+  }, 1500);
+
+  searchHandler = (newCityInput: AppState['newCityInput']) => {
+    this.setState({ newCityInput, isWeatherLoading: true, weatherErr: false, error: '' });
+    this.debouncedSearch(newCityInput);
+  }
+
+  toggleUnitsHandler =  (unit: AppState['unit']) => {
+    this.setState({ unit })
+    this.getWeatherData(this.state.newCityInput || this.state.city, unit);
+  };
+
+  getWeatherData = (city: AppState['city'], unit: AppState['unit']) => {
+    let urlParams = new URLSearchParams({ q: city, appid: `${process.env.REACT_APP_APPID}`, units: unit }).toString();
+    fetch(`https://api.openweathermap.org/data/2.5/weather?${urlParams}`)
+      .then(weather => {
+        if (weather.ok) {
+          return weather.json()
+        }  else {
+          console.log(weather);
+          if (weather.status === 404) {
+            this.setState({ error: '404'});
+          }
+          throw new Error('Network err')
+        }})
+      .then(weatherInfo => {
+        this.setState( { weatherInfo, isWeatherLoading: false, error: ''} )
+      })
+      .catch(() => {
+        this.setState( { weatherErr: true, isWeatherLoading: false, weatherInfo: null})
+      })
+  }
+
+  componentDidMount() {
+    const city = localStorage.getItem(this.localStorageKey) ?? '[]';
+    if (city.length) {
+      this.setState({ city });
+    }
+    if (this.state.city === '' && this.state.isLocationLoading){
+      fetch('http://ip-api.com/json/')
+        .then(location => {
+          if (location.ok) {
+            return location.json()
+          } else {
+            return new Error('Network error');
+          }})
+        .then(location => {
+          this.setState( { city: location.city, isLocationLoading: false } );
+          localStorage.setItem(this.localStorageKey, this.state.city);
+          this.getWeatherData(location.city, this.state.unit);
+        })
+        .catch(() => { this.setState( { city: 'Minsk', locationErr: true , isLocationLoading: false} )})
+    }
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <header className={ css.headerBlock }>
+          <div className={ css.container }>
+            <Input placeholder={ 'Weather in city' } value={ this.state.newCityInput } onChange={ (e) => this.searchHandler(e.target.value.trim()) }/>
```

Comment:
строки - часть html, так что для них не обязательный фигурные скобки
```suggestion
            <Input placeholder="Weather in city" value={ this.state.newCityInput } onChange={ (e) => this.searchHandler(e.target.value.trim()) }/>
```